store/mongo/datastore_mongo.go

```diff
@@ -535,6 +535,71 @@ func (db *DataStoreMongo) UpdateDevicesGroup(
 	}, nil
 }
 
+func (db *DataStoreMongo) GetFiltersAttributes(ctx context.Context) ([]model.FilterAttribute, error) {
+	database := db.client.Database(mstore.DbFromContext(ctx, DbName))
+	collDevs := database.Collection(DbDevicesColl)
+
+	const DbCount = "count"
+
+	cur, err := collDevs.Aggregate(ctx, []bson.M{
+		{
+			"$project": bson.M{
+				"attributes": bson.M{
+					"$objectToArray": "$" + DbDevAttributes,
+				},
+			},
+		},
+		{
+			"$unwind": "$" + DbDevAttributes,
+		},
+		{
+			"$project": bson.M{
+				DbDevAttributesName:  "$" + DbDevAttributes + ".v." + DbDevAttributesName,
+				DbDevAttributesScope: "$" + DbDevAttributes + ".v." + DbDevAttributesScope,
+			},
+		},
+		{
+			"$group": bson.M{
+				DbDevId: bson.M{
+					DbDevAttributesName:  "$" + DbDevAttributesName,
+					DbDevAttributesScope: "$" + DbDevAttributesScope,
+				},
+				DbCount: bson.M{
+					"$sum": 1,
+				},
+			},
+		},
+		{
+			"$sort": bson.D{
+				{Key: DbCount, Value: -1},
+				{Key: DbDevId + "." + DbDevAttributesScope, Value: 1},
+				{Key: DbDevId + "." + DbDevAttributesName, Value: 1},
+			},
+		},
+	})
+	if err != nil {
+		return nil, err
+	}
+	defer cur.Close(ctx)
+
+	var results []bson.M
+	err = cur.All(ctx, &results)
+	if err != nil {
+		return nil, err
+	}
+
+	var attributes []model.FilterAttribute
+	for _, item := range results {
+		attributes = append(attributes, model.FilterAttribute{
+			Name:  item[DbDevId].(bson.M)[DbDevAttributesName].(string),
+			Scope: item[DbDevId].(bson.M)[DbDevAttributesScope].(string),
+			Count: item[DbCount].(int32),
+		})
+	}
+
```

Comment:
This is not ideal.
If you reformat the `$group` stage in the pipeline you can decode this directly from the cursor.

Reply:
@alfrunes thanks for the suggestion, you are right